hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleContext.java

```diff
@@ -262,12 +260,36 @@ default SignatureVerification verificationFor(@NonNull Key key, @NonNull Verific
      * aware that any state changes introduced by storing data in one of the stores after calling a child transaction
      * will also be rolled back if the child transaction is rolled back.
      *
+     * <p>The provided {@link VerificationAssistant} callback will be called when the child transaction calls any
+     * of the {@code verificationFor} methods. If both, parent and child transaction, provide a
+     * {@link VerificationAssistant}, the one from the parent will be called first.
+     *
      * <p>A {@link TransactionCategory#PRECEDING}-transaction must not dispatch a child transaction.
      *
      * @param txBody the {@link TransactionBody} of the child transaction to dispatch
      * @param recordBuilderClass the record builder class of the child transaction
+     * @param callback a {@link VerificationAssistant} callback function that will observe each primitive key
      * @return the record builder of the child transaction
-     * @throws NullPointerException if {@code txBody} is {@code null}
+     * @throws NullPointerException if any of the arguments is {@code null}
+     * @throws IllegalArgumentException if the current transaction is a
+     * {@link TransactionCategory#PRECEDING}-transaction or if the record builder type is unknown to the app
+     */
+    @NonNull
+    <T> T dispatchChildTransaction(
+            @NonNull TransactionBody txBody,
+            @NonNull Class<T> recordBuilderClass,
```

Comment:
This parameter is still unclear.  The caller does not necessarily know the type of the transaction, so how does it provide a record builder class?  Even if it knew the type, that would make the calling service (e.g. Schedule Service) directly dependent on the implementation of every other service (to obtain the record builder class), which seems less than ideal.

Reply:
For cases where the caller does not know the type of transaction, I introduced the new interface `SingleTransactionRecordBuilder` which will make general properties available. But in other cases we know the dispatched transaction and may need a more detailed answer.

Reply:
For scheduled transactions we need to pass data *in* to the child, as the record in the record stream for a scheduled transaction has a `scheduleRef` field with the `ScheduleID` of the schedule that managed its execution.

Reply:
I think in that case, you have to define an interface that allows you to set the `scheduleRef` and `SingleTransactionRecordBuilder` to implement it. You can use it as the `recordBuilderClass` and set the `scheduleRef` when the call returns.